Unsupported properties file format via `MICRONAUT_CONFIG_FILES` when using GradleUpShadow 9

### Expected Behavior

When running a JAR built with **com.gradleup.shadow** `9.x.x` and the `MICRONAUT_CONFIG_FILES` environment variable set to a YAML config file, it should be able to read this file.

### Actual Behaviour

Currently, the JAR fails at startup with the following error:

```java
io.micronaut.context.exceptions.ConfigurationException: Unsupported properties file format while reading application-dev.yaml from D:\Projects\test\src\main\resources\application-dev.yaml
        at io.micronaut.context.env.DefaultEnvironment.lambda$readPropertySourceListFromFiles$4(DefaultEnvironment.java:527)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.micronaut.context.env.DefaultEnvironment.readPropertySourceListFromFiles(DefaultEnvironment.java:511)
        at io.micronaut.context.env.DefaultEnvironment.readPropertySources(DefaultEnvironment.java:450)
        at io.micronaut.context.env.DefaultEnvironment.start(DefaultEnvironment.java:280)
        at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:979)
        at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:947)
        at io.micronaut.context.DefaultApplicationContext.startEnvironment(DefaultApplicationContext.java:295)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:224)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:347)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:333)
        at com.example.Application.main(Application.java:8)
```

When using **com.gradleup.shadow** `9.0.0`, it fails with the following error instead:

```java
java.lang.NoClassDefFoundError: org/yaml/snakeyaml/constructor/BaseConstructor
        at io.micronaut.context.env.yaml.YamlPropertySourceLoader.processInput(YamlPropertySourceLoader.java:59)
        at io.micronaut.context.env.AbstractPropertySourceLoader.read(AbstractPropertySourceLoader.java:148)
        at io.micronaut.context.env.DefaultEnvironment.readPropertiesFromLoader(DefaultEnvironment.java:676)
        at io.micronaut.context.env.DefaultEnvironment.lambda$readPropertySourceListFromFiles$4(DefaultEnvironment.java:521)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.micronaut.context.env.DefaultEnvironment.readPropertySourceListFromFiles(DefaultEnvironment.java:511)
        at io.micronaut.context.env.DefaultEnvironment.readPropertySources(DefaultEnvironment.java:450)
        at io.micronaut.context.env.DefaultEnvironment.start(DefaultEnvironment.java:280)
        at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:979)
        at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:947)
        at io.micronaut.context.DefaultApplicationContext.startEnvironment(DefaultApplicationContext.java:295)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:224)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:347)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:333)
        at com.example.Application.main(Application.java:8)
Caused by: java.lang.ClassNotFoundException: org.yaml.snakeyaml.constructor.BaseConstructor
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        ... 16 common frames omitted
```

The error changes to the first one starting precisely from **com.gradleup.shadow** `9.0.1`, but I didn’t notice anything particularly relevant in the release notes.

The [Diffuse](https://github.com/JakeWharton/diffuse) analysis comparing versions `v8.3.x` and `v9.2.2` is as follows:

```bash
 JAR   │ old       │ new       │ diff
───────┼───────────┼───────────┼──────────
 class │  25,2 MiB │  25,2 MiB │ -1,2 KiB
 other │ 409,5 KiB │ 408,2 KiB │ -1,3 KiB
───────┼───────────┼───────────┼──────────
 total │  25,6 MiB │  25,6 MiB │ -2,5 KiB

 CLASSES │ old   │ new   │ diff
─────────┼───────┼───────┼────────────
 classes │  7014 │  7013 │ -1 (+0 -1)
 methods │ 71798 │ 71798 │  0 (+0 -0)
  fields │ 26769 │ 26769 │  0 (+0 -0)

=================
====   JAR   ====
=================

 size    │ diff     │ path
─────────┼──────────┼──────────────────────────────────────────────────────────────────────────
         │   -484 B │ - META-INF/versions/11/module-info.class
         │   -748 B │ - META-INF/versions/9/module-info.class
     0 B │      0 B │ + META-INF/services/
         │      0 B │ - META-INF/versions/9/
    83 B │    +21 B │ ? META-INF/MANIFEST.MF
    60 B │   -449 B │ ? META-INF/services/io.micronaut.core.convert.TypeConverterRegistrar
   265 B │   -111 B │ ? META-INF/services/io.micronaut.context.env.PropertySourceLoader
    50 B │   -124 B │ ? META-INF/services/io.micronaut.core.type.TypeInformationProvider
   668 B │   -679 B │ ? META-INF/services/reactor.blockhound.integration.BlockHoundIntegration
─────────┼──────────┼──────────────────────────────────────────────────────────────────────────
 1,1 KiB │ -2,5 KiB │ (total)
```

What did I miss?

### Steps To Reproduce

1. Create a new Micronaut project.
2. Update the **com.gradleup.shadow** plugin to the latest version (`9.2.2`).
3. Run `gradlew.bat build`
4. Export the YAML config file using the `MICRONAUT_CONFIG_FILES` environment variable:

```bash
set MICRONAUT_CONFIG_FILES=D:\Projects\test\src\main\resources\application-dev.yml
```

5. Run the application:

```bash
java -jar demo-0.1-all.jar
```

### Environment Information

**com.gradleup.shadow**: 9.2.2
**Micronaut**: 4.9.4
**Micronaut Gradle plugin**: 4.5.5

### Example Application

https://github.com/loicgreffier/unsupported-properties-file

### Version

4.9.4

--- a/minimal-plugin/src/main/java/io/micronaut/gradle/MicronautComponentPlugin.java
+++ b/minimal-plugin/src/main/java/io/micronaut/gradle/MicronautComponentPlugin.java
@@ -93,7 +93,7 @@ public class MicronautComponentPlugin implements Plugin<Project> {
 
         ShadowPluginSupport.withShadowPlugin(project, () -> {
             configureTesting(project, micronautExtension, inspectRuntimeClasspath);
-            ShadowPluginSupport.mergeServiceFiles(project);
+            ShadowPluginSupport.configureDefaults(project);
         });
         PluginsHelper.registerVersionExtensions(PluginsHelper.KNOWN_VERSION_PROPERTIES, project);
 

--- a/minimal-plugin/src/main/java/io/micronaut/gradle/MicronautMinimalApplicationPlugin.java
+++ b/minimal-plugin/src/main/java/io/micronaut/gradle/MicronautMinimalApplicationPlugin.java
@@ -186,7 +186,7 @@ public class MicronautMinimalApplicationPlugin implements Plugin<Project> {
             if (micronautRuntime == MicronautRuntime.GOOGLE_FUNCTION) {
                 configureGoogleCloudFunctionRuntime(project, p, dependencyHandler);
             }
-            ShadowPluginSupport.withShadowPlugin(project, () -> ShadowPluginSupport.mergeServiceFiles(project));
+            ShadowPluginSupport.withShadowPlugin(project, () -> ShadowPluginSupport.configureDefaults(project));
 
         });
     }

--- a/minimal-plugin/src/main/java/io/micronaut/gradle/ShadowPluginSupport.java
+++ b/minimal-plugin/src/main/java/io/micronaut/gradle/ShadowPluginSupport.java
@@ -17,6 +17,7 @@ package io.micronaut.gradle;
 
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar;
 import org.gradle.api.Project;
+import org.gradle.api.file.DuplicatesStrategy;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -58,8 +59,15 @@ public class ShadowPluginSupport {
      * <a href="https://imperceptiblethoughts.com/shadow/configuration/merging/#merging-service-descriptor-files">Shadow: Merging Server Descriptor Files</a>
      * @param project Gradle Project
      */
-    public static void mergeServiceFiles(Project project) {
+    private static void mergeServiceFiles(Project project) {
         project.getTasks().withType(ShadowJar.class).configureEach(ShadowPluginSupport::mergeServiceFiles);
+    }
+
+    public static void configureDefaults(Project project) {
+        project.getTasks().withType(ShadowJar.class).configureEach(jar -> {
+            jar.setDuplicatesStrategy(DuplicatesStrategy.WARN);
+            mergeServiceFiles(jar);
+        });
     }
 
     // This method calls `mergeServiceFiles` reflectively, because the Shadow Plugin v9

